drivers/ethernet/eth_xilinx_axienet.c

```diff
@@ -0,0 +1,618 @@
+		}
+		if (link_state.speed & LINK_HALF_1000BASE_T ||
+		    link_state.speed & LINK_FULL_1000BASE_T) {
+			config->l.link_1000bt = true;
+		}
+		return 0;
+	case ETHERNET_CONFIG_TYPE_MAC_ADDRESS:
+		memcpy(config->mac_address.addr, data->mac_addr, sizeof(data->mac_addr));
+		return 0;
+	case ETHERNET_CONFIG_TYPE_PROMISC_MODE:
+		/* not supported yet */
+		config->promisc_mode = false;
+		return 0;
+	case ETHERNET_CONFIG_TYPE_RX_CHECKSUM_SUPPORT:
+		if (dev_config->have_rx_csum_offload) {
+			config->chksum_support =
+				ETHERNET_CHECKSUM_SUPPORT_IPV4_HEADER |
+				ETHERNET_CHECKSUM_SUPPORT_TCP | ETHERNET_CHECKSUM_SUPPORT_UDP |
+				ETHERNET_CHECKSUM_SUPPORT_IPV6_HEADER |
+				ETHERNET_CHECKSUM_SUPPORT_TCP | ETHERNET_CHECKSUM_SUPPORT_UDP;
+		} else {
+			config->chksum_support = ETHERNET_CHECKSUM_SUPPORT_NONE;
+		}
+		return 0;
+	case ETHERNET_CONFIG_TYPE_TX_CHECKSUM_SUPPORT:
+		if (dev_config->have_tx_csum_offload) {
+			config->chksum_support =
+				ETHERNET_CHECKSUM_SUPPORT_IPV4_HEADER |
+				ETHERNET_CHECKSUM_SUPPORT_TCP | ETHERNET_CHECKSUM_SUPPORT_UDP |
+				ETHERNET_CHECKSUM_SUPPORT_IPV6_HEADER |
+				ETHERNET_CHECKSUM_SUPPORT_TCP | ETHERNET_CHECKSUM_SUPPORT_UDP;
+		} else {
+			config->chksum_support = ETHERNET_CHECKSUM_SUPPORT_NONE;
+		}
+		return 0;
+	default:
+		LOG_ERR("Unsupported configuration queried: %u", type);
+		return -EINVAL;
+	}
+}
+
+static void xilinx_axienet_iface_init(struct net_if *iface)
+{
+	struct xilinx_axienet_data *data = net_if_get_device(iface)->data;
+
+	data->interface = iface;
+
+	ethernet_init(iface);
+
+	net_if_set_link_addr(iface, data->mac_addr, sizeof(data->mac_addr), NET_LINK_ETHERNET);
+
+	LOG_INF("Interface initialized!");
+}
+
+static int xilinx_axienet_send(const struct device *dev, struct net_pkt *pkt)
+{
+	struct xilinx_axienet_data *data = dev->data;
+	const struct xilinx_axienet_config *config = dev->config;
+	size_t pkt_len = net_pkt_get_len(pkt);
+	size_t current_descriptor = data->tx_populated_buffer_index;
+
+	if (net_pkt_read(pkt, data->tx_buffer[current_descriptor].buffer, pkt_len)) {
+		LOG_ERR("Failed to read packet into TX buffer!");
+		return -EIO;
+	}
+	return setup_dma_tx_transfer(dev, config, data, pkt_len);
+}
+
+static void xilinx_axienet_set_mac_address(const struct xilinx_axienet_config *config,
+					   const struct xilinx_axienet_data *data)
+{
+	xilinx_axienet_write_register(config, XILINX_AXIENET_UNICAST_ADDRESS_WORD_0_OFFSET,
+				      (data->mac_addr[0]) | (data->mac_addr[1] << 8) |
+					      (data->mac_addr[2] << 16) |
+					      (data->mac_addr[3] << 24));
+	xilinx_axienet_write_register(config, XILINX_AXIENET_UNICAST_ADDRESS_WORD_1_OFFSET,
+				      (data->mac_addr[4]) | (data->mac_addr[5] << 8));
+}
+
+static void phy_link_state_changed(const struct device *dev, struct phy_link_state *state,
+				   void *user_data)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(user_data);
+
+	LOG_INF("Link state changed to: %s (speed %x)", state->is_up ? "up" : "down", state->speed);
```

Comment:
consider adding something like
```
	if (state->is_up) {
		net_eth_carrier_on(context->iface);
	} else {
		net_eth_carrier_off(context->iface);
	}
```
to inform the network subsystem, of the link state.
you should then also do ``net_if_carrier_off(iface);`` before the ``phy_link_callback_set()``

Reply:
Done